sw/otbn/code-snippets/randomness.s

```diff
@@ -111,11 +110,38 @@ test_urnd:
   /* Read the URND WSR again. */
   bn.wsrr w1, 0x2 /* URND */
 
-  /* Fail the test if the two numbers read from URND are equal. */
+  jal x0, error_checking
+
+/**
+ * Perform the following verifications:
+ *
+ *     w0 != w1  - The two random number read shall be different   
+ *     w0 != 0   - The random number shall not have all bits equal to zero 
+ *     w0 != ~0  - The random number shall not have all bits equal to one 
+ */ 
+error_checking:
+  /* Fail the test if the two numbers read from RND are equal. */
   bn.cmp w0, w1
-  csrrs x2, 0x7c0 /* FG0 */, x0
+  csrrs x2, CSR_FG0, x0
+  andi x2, x2, 8 /* filter out Z (zero) bit */
```

Comment:
Comments should be complete sentences. (I.e., start with a capital letter, end with a period.)

Reply:
Thanks! Done!